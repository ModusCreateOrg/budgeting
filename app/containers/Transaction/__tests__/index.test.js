import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Transaction } from '../index';

// mock nested component
jest.mock('components/NavLink', () => ({ to, label }) => <a href={to}>{label}</a>);
jest.mock('components/Transaction', () => ({ description, value }) => (
  <article>
    <h1>{description}</h1>
    <h2>{value}</h2>
  </article>
));

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Transaction transaction={{ description: 'Test', value: 10 }} balance={10} />
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
