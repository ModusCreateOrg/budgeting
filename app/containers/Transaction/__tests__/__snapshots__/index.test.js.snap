// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <a
    href="/budget"
  >
    ‚Üê Budget
  </a>
  <article>
    <h1>
      Test
    </h1>
    <h2>
      10
    </h2>
  </article>
</section>
`;
